### PR TITLE
Init package

### PR DESCRIPTION
**Pull request by AdityaRajSingh**

### CODE REVIEW

File: presigned-post.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	Bucket          string // AWS S3 bucket.
+	AccessKeyID     string // AWS access key.
+	SecretAccessKey string // AWS secret access key.
+}
+
+// GetCredentials from the environment.
+func GetCredentials() (*Credentials, error) {
+	// Get AWS credentials from environment variables.
+	region := os.Getenv("AWS_REGION")
+	bucket := os.Getenv("AWS_S3_USER_ASSET_BUCKET")
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: use a generic name instead of AWS_S3_USER_ASSET_BUCKET

File: presigned-post.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    {"x-amz-credential": "%s"},
+    {"x-amz-algorithm": "%s"},
+    {"x-amz-date": "%s" },
+    {"Content-Type": "%s"},
+    {"Cache-Control": "%s"},
+  ]
+}
+`
+
+// Policy represents a new policy for uploading sounds.
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: Fix the comment

File: presigned-post.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	Bucket       string         // AWS S3 bucket.
+	Key          string         // Key (file path) for the S3 object.
+	Acl          string         // AWS S3 ACL (Access Control List).
+	Credential   string         // AWS credential information.
+	Algorithm    string         // AWS algorithm for the policy.
+	Date         string         // Date of the policy.
+	C            *Credentials   // AWS credentials.
+	O            *PolicyOptions // Policy options.
+	ContentType  string         // Content type of the S3 object.
+	CacheControl string         // Cache control header.
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: Will there be PolicyConditions type ?

File: presigned-post.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+// String returns the policy as a string.
+func (p *Policy) String() string {
+	// Format the policy using the PolicyTemplate.
+	s1 := fmt.Sprintf(PolicyTemplate,
+		p.Expiration,
+		p.Bucket,
+		p.Key,
+		p.Acl,
+		p.O.MaxFileSizeInBytes,
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: if p.O is optional, then it can break

File: presigned-post.go [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    {"Cache-Control": "%s"},
+  ]
+}
+`
+
+// Policy represents a new policy for uploading sounds.
+type Policy struct {
+	Expiration   string         // Expiration time of the policy.
+	Region       string         // AWS region.
+	Bucket       string         // AWS S3 bucket.
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: Bucket is present in Credential struct as well

File: presigned-post.go [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+// PolicyOptions represent policy options.
+type PolicyOptions struct {
+	ExpiryInMinutes    int // Expiration time in minutes for the policy.
+	MaxFileSizeInBytes int // Maximum allowed file size in the policy.
+}
+
+// PolicyConditions represent policy conditions.
+type PolicyConditions struct {
+	Acl          string // AWS S3 ACL (Access Control List).
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: Add more comments on what are the expected values

File: presigned-post.go [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+`
+
+// Policy represents a new policy for uploading sounds.
+type Policy struct {
+	Expiration   string         // Expiration time of the policy.
+	Region       string         // AWS region.
+	Bucket       string         // AWS S3 bucket.
+	Key          string         // Key (file path) for the S3 object.
+	Acl          string         // AWS S3 ACL (Access Control List).
+	Credential   string         // AWS credential information.
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: Are there any enum types in these keys. Also which options are optional?

File: presigned-post.go [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+		Bucket:          bucket,
+		AccessKeyID:     accessKeyID,
+		SecretAccessKey: secretAccessKey,
+	}, nil
+}
+
+// PolicyOptions represent policy options.
+type PolicyOptions struct {
+	ExpiryInMinutes    int // Expiration time in minutes for the policy.
+	MaxFileSizeInBytes int // Maximum allowed file size in the policy.
```
Comment (AMAN-BARBARIA) [inline presigned-post.go]: What is the difference between PolicyOptions & PolicyConditions?

File: presigned-post.go [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+	"encoding/hex"
+	"fmt"
+	"os"
+	"time"
+)
+
+// Credentials represent AWS credentials and config.
+type Credentials struct {
+	Region          string // AWS region.
+	Bucket          string // AWS S3 bucket.
```
Comment (AMAN-BARBARIA) [inline presigned-post.go:15]: Bucket should be outside Credentials struct?

**Review by AMAN-BARBARIA — COMMENTED** [review]
Added a few comments

File: presigned-post.go (hunk 3 above)
Comment (AdityaRajSingh) [reply presigned-post.go]: Nope, also removed Policy Options as well.

File: presigned-post.go (hunk 5 above)
Comment (AdityaRajSingh) [reply presigned-post.go]: Yes it is required at both the places.

**Review by sunilkhedar — APPROVED** [review]
LGTM

**Pull request merged**